OIPA/iati/parser/tests/test_2_03_parser.py

```diff
@@ -1658,6 +1658,126 @@ def test_activity_result_document_link_language(self):
+            root=self.iati_203_XML_file,
+        ).get_parser()
+
+        # Related objects:
+        self.result = iati_factory.ResultFactory.create()
+        self.activity = self.result.activity
+
+        self.parser_203.register_model('Activity', self.activity)
+        self.parser_203.register_model('Result', self.result)
+
+    def test_activity_result_indicator(self):
+        """test if <indicator> element inside <result> element is parsed
+        properly
+        """
+
+        # 1) measure attribute is missing:
+        result_indicator_attrs = {
+            # "measure": "1",
+            "ascending": "1",
+            "aggregation-status": "1",
+
+        }
+
+        result_indicator_XML_element = E(
+            'indicator',
+            **result_indicator_attrs
+        )
+
+        try:
+            self.parser_203.\
+                iati_activities__iati_activity__result__indicator(
+                    result_indicator_XML_element
+                )
+            self.assertFail()
+        except RequiredFieldError as inst:
+            self.assertEqual(inst.field, 'measure')
+            self.assertEqual(inst.message, 'required attribute missing')
+
+        # 2) IndicatorMeasure codelist is missing:
+
+        result_indicator_attrs = {
+            "measure": "1",
+            "ascending": "1",
+            "aggregation-status": "1",
+
+        }
+
+        result_indicator_XML_element = E(
+            'indicator',
+            **result_indicator_attrs
+        )
+
+        try:
+            self.parser_203.\
+                iati_activities__iati_activity__result__indicator(
+                    result_indicator_XML_element
+                )
+            self.assertFail()
+        except RequiredFieldError as inst:
+            self.assertEqual(inst.field, 'measure')
+            self.assertEqual(
+                inst.message,
+                'not found on the accompanying code list'
+            )
+
+        # 3) everything's OK:
+        # let's create an IndicatorMeasure codelist object:
+
+        self.parser_203.codelist_cache = {}
+
+        result_indicator_attrs = {
+            "measure": "1",
+            "ascending": "1",
+            "aggregation-status": "1",
+
+        }
+
+        indicator_measure = iati_factory.IndicatorMeasureFactory.create(
+            code=result_indicator_attrs.get("measure")
+        )
+
+        result_indicator_XML_element = E(
+            'indicator',
+            **result_indicator_attrs
+        )
+
+        self.parser_203.\
+            iati_activities__iati_activity__result__indicator(
+                result_indicator_XML_element
+            )
+
+        result_indicator = self.parser_203.get_model('ResultIndicator')
+
+        self.assertEqual(result_indicator.result, self.result)
+        self.assertEqual(
+            result_indicator.measure,
+            indicator_measure
+        )
+        self.assertEqual(result_indicator.ascending, True)
```

Comment:
what if 'ascending' attribute is 'False' ?

Reply:
Well `makeBool()` function will then return `False` :) But you're right - this `makeBool()` function needs an update and  its separate test too, that would check all possible true / false values (based on IATI docs) and that would raise an error if any _other_ value than allowed would be passed. But I don't think this is in the scope of this test to be honest.

Reply:
I can also see that this function not tested at all, I will at least add a `TODO: test comment`